clang/test/Analysis/asm.cpp

```diff
@@ -40,7 +42,19 @@ void testInlineAsmMemcpyUninit(void)
 {
     int a[10], b[10] = {}, c;
     MyMemcpy(&a[1], &b[1], sizeof(b) - sizeof(b[1]));
-    c = a[0]; // expected-warning{{Assigned value is garbage or undefined}}
+    c = a[0]; // FIXME: should be warning about uninitialized value, but invalidateRegions() also
```

Comment:
This is unfortunate, but not really sure how to avoid it

Reply:
You can set the invalidation traint to preserve super region.

Reply:
Don't think it would work, since then `testInlineAsmMemcpyUninitLoop` would also fail, since only the first element of the array will be invalidated.

Maybe I am missing some API to invalidate part of the region?

Reply:
On the other hand, I think, this behavior maybe even better in terms of FP. Like maybe asm block will do smth like `container_of` or such. 

So this might be an acceptable damage

Reply:
I suppose it all boils down to what the assembly block does. I don't think there is anything fundamentally preventing us refining what gets invalidated other than actually parsing and understanding the asm block.

Invalidating everything is a good default choice; however, frequently we can do better than that.
IDK how the `MyMemcpy` works in your test, but it could model the invalidation similar to how the StreamChecker goes smart about it in `tryToInvalidateFReadBufferByElements` [here](https://github.com/llvm/llvm-project/blob/2e3c7dbbcbfa37ae83251bb3da388df772680689/clang/lib/StaticAnalyzer/Checkers/StreamChecker.cpp#L1108-1147).